vortex-array/src/data/mod.rs

```diff
@@ -399,6 +399,15 @@ impl<T: AsRef<ArrayData>> ArrayDType for T {
     }
 }
 
+impl ArrayData {
+    pub fn into_dtype(self) -> DType {
```

Comment:
Is this worth it?

Reply:
The cost seems small both conceptually (it's just the into_ version of dtype) and technically (not much code in the binary or the source). I think my instinct is that DType construction is kinda expensive, particularly if you have thousands of columns.

Reply:
So this is useful for empty ArrayData used in the dtype below?

Reply:
yeah, I imagine it's useful for computing the dtype of an empty array and then taking ownership of the empty array's dtype.

Reply:
I think I'd rather we _don't_ do this, and if it does turn out to be expensive, then we should Arc nested dtypes, e.g. `DType::Struct(Arc<StructDType>)`